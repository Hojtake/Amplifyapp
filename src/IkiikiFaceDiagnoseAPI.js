export default class IkiikiFaceDiagnoseAPI {
    async callLoginAPI (ID, password) {
        const myHeaders = new Headers();
        myHeaders.append("Content-Type", "application/json");
       
        const requestOptions = {
            method: 'POST',
            headers: myHeaders,
            body: JSON.stringify({ "ID": ID, "password": password }),
            redirect: 'follow'
        };
        const response = await fetch("https://nczxmo91g3.execute-api.ap-northeast-1.amazonaws.com/login", requestOptions);
        return response.json();
    }

    async callFaceDiagnoseAPI (Image, ID) {
        const myHeaders = new Headers();
        myHeaders.append("Content-Type", "application/json");
        const requestOptions = {
            method: 'POST',
            headers: myHeaders,
            body: JSON.stringify({ "Image": Image, "ID": ID }),
            redirect: 'follow'
        };
        const response = await fetch("https://sdnl1xmao1.execute-api.ap-northeast-1.amazonaws.com/ikiiki-value", requestOptions);
        return response.json();
    }
}