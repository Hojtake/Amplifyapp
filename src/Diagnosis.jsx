
import React from "react";
import ReactDOM from "react-dom";
import FunctionSelection  from "./FunctionSelection.jsx";
import IkiikiFaceDiagnoseAPI from "./IkiikiFaceDiagnoseAPI";
import classes from "./Diagnosis.module.css";
export default class Diagnosis extends React.Component {
    constructor(props){
        super(props);
        this.clickReturnToFunctionSelection = this.clickReturnToFunctionSelection.bind(this);
        this.clickDiagnose = this.clickDiagnose.bind(this);
        this.clickImageSelect = this.clickImageSelect.bind(this);
        this.state={resistDayMessage:null,resultMessage:null,operationMessage:"・画像選択ボタンを押してください",photoimage:""};
    }
    //画像をアップロードしたときに画像表示領域に画像を表示する関数
    clickImageSelect= (e)=>{
        
        this.setState({resultMessage:null});
        this.setState({resistDayMessage:null});
        const filelist = e.target.files;
        if(filelist.length == 0) return;
        this.setState({photoimage:""});
        const reader = new FileReader();
        const faceImage = new Image();
        const file = filelist[0];
        reader.onload = function (e) {
            faceImage.onload = function () {
                //画像表示領域の横幅(px)
                const windowWidth = 700;
                //画像表示領域の縦幅(px)
                const windowHeight =400;
                //画像表示領域の縦横比
                const windowRatio=windowHeight/windowWidth;
                //表示する画像の横幅
                let imageWidth;
                //表示する画像の縦幅
                let imageHeight;
                const ratio = faceImage.height / faceImage.width;
                //縦横比に応じて画面表示領域にフィットするようにimgのwidth,heightを決定する。
                if (ratio > windowRatio) {
                    //画像の縦幅を画像表示領域に合わせてフィットさせる
                    imageHeight = windowHeight;
                    imageWidth =faceImage.width * windowHeight / faceImage.height;
                }
                else {
                    //画像の横幅を画像表示領域に合わせてフィットさせる
                    imageHeight = faceImage.height * windowWidth / faceImage.width;
                    imageWidth=windowWidth;
                }
                //画像を画像表示領域内に埋め込む
                const imgdata = <img src={e.target.result} width={imageWidth} height={imageHeight} id="getimg"/>;
                this.setState({photoimage:imgdata})
            }.bind(this)
            faceImage.src= e.target.result;
        }.bind(this)
        reader.readAsDataURL(file);
        this.setState({operationMessage:"・診断するボタンを押してください"});
    }

    clickReturnToFunctionSelection = ()=>{
        ReactDOM.render(<FunctionSelection ID={this.props.ID}/>,document.getElementById("root"));
    }

    clickDiagnose = ()=>{
    	this.setState({resistDayMessage:null});
    	this.setState({resultMessage:null});
        const image = document.getElementById("getimg");
        if(image == null){
            this.setState({resultMessage:"画像を選択してから診断するボタンを押してください。"});
            return ;    
        }
        
        const api = new IkiikiFaceDiagnoseAPI();
        api.callFaceDiagnoseAPI(image.getAttribute("src"),this.props.ID)
        .then(result =>{
            //result及びresult内部のパラメータがnullまたはundefinedの場合にエラーとして処理を行う
            if(!result || !result.message){
                throw new Error();
                }
            if(result.hasFaceDiagnosed){
                if(!result.date || result.ikiikiValue === null || result.ikiikiValue === ""){
                    throw new Error();
                }
                this.setState({resistDayMessage:`${result.date}本日のイキイキ度は${result.ikiikiValue}です。`});
                this.setState({resultMessage:`${result.message}`});
            }else{                     
                this.setState({resultMessage:result.message});
            }                    
        }).catch(()=>{
            this.setState({resultMessage:"予期しないエラーが発生しました。しばらく待ってから再度実行してください。"});
        });           
    }

    render(){        
        return (
            <>
                <h1>イキイキ顔診断画面</h1>
                <div className={classes.username}><p id="ID">ID:{this.props.ID}</p></div>
                <div className={classes.return_to_function_select_area} ><button onClick={this.clickReturnToFunctionSelection} id="clickreturn">機能選択画面に戻る</button></div>
                <div className={classes.message_area}><p id="operationMsg">{this.state.operationMessage}</p></div>
                <div className={classes.photo_area} id="photo_area">{this.state.photoimage}</div>
                <div className={classes.button_area}>
                    <label htmlFor="filename" className={classes.label}>画像を選択<input type="file" id="filename" accept=".png,.jpg,.jpeg" onChange={this.clickImageSelect}/></label>    
                    <button className={classes.diagnose_button} onClick={this.clickDiagnose} id="diagnose_button">診断する</button>
                </div>
                <div className={classes.diagnose_result_area} id="diagnose_result_area">
                    <p id="resist_day">{this.state.resistDayMessage}</p>
                     <p id="result_msg">{this.state.resultMessage}</p>
                </div>
            </>
        );
    }
}