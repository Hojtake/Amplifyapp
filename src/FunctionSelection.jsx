import React from "react";
import ReactDOM from "react-dom";
import Diagnosis from "./Diagnosis";
import Login from "./Login";
import classes from"./FunctionSelection.module.css";

export default class FunctionSelection extends React.Component{
    constructor(props){
        super(props);
        this.clickLogout = this.clickLogout.bind(this);
        this.clickIkiikiDiagnoseFace = this.clickIkiikiDiagnoseFace.bind(this);
    }
    clickLogout = () =>{
        ReactDOM.render(<Login/>,document.getElementById("root"));
    }
    clickIkiikiDiagnoseFace = () => {
        ReactDOM.render(<Diagnosis ID={this.props.ID} ikiikiResults={this.props.ikiikiResults}/>,document.getElementById("root"));
    }

    render(){
        let errMsg=null;
        let ikiikiData=[]; 
        let noDataMsg= null
        if(this.props.hasReadData){
            if(this.props.ikiikiResults.length > 0){
                for(let i=0;i <this.props.ikiikiResults.length;i++){
                    let styleData;
                    let diagnosedData = this.props.ikiikiResults[i];
                    if(diagnosedData["ikiikiValue"]<50){
                        styleData="rgb(255, 125, 125)";
                    }else{
                        styleData="transparent";
                    }
                    ikiikiData.push(<tr key={i} style={{backgroundColor:styleData}}><td>{diagnosedData["diagnosedDate"]}</td><td>{diagnosedData["ikiikiValue"]}</td></tr>);
                
                }
            }else{
                noDataMsg = "まだイキイキ度が登録されていません。";
            }
        }
        return(
            <>
                <h1>機能選択画面</h1>
                <p className={classes.error_message}>{errMsg}</p>
                <div className={classes.username}><p>ID:{this.props.ID}</p></div>
                <div className={classes.logout_button}><button onClick={this.clickLogout} id="logout">ログアウト</button></div>
                <div className={classes.function_area}>
                <button className={classes.function} onClick={this.clickIkiikiDiagnoseFace} id="diagnoseFace">イキイキ顔診断</button>
                </div>
                <div className={classes.table}>
                <div className={classes.table_area}>
                <p>{this.props.ID}さんの過去のイキイキ度の記録</p>
                <p>{noDataMsg}</p>
                    <table border="2" width="650">
                    
                    <thead>
                    <tr>
                        <th width="50%">登録日</th>
                        <th>イキイキ度</th>
                    </tr>
                    </thead>
                    
                    <tbody>{ikiikiData}</tbody>
                    </table></div>
                    </div>
            </>
        )
    }

}