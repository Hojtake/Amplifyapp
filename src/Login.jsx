import React from "react";
import ReactDOM from "react-dom";
import FunctionSelection from "./FunctionSelection";
import IkiikiFaceDiagnoseAPI from "./IkiikiFaceDiagnoseAPI";
import classes from "./Login.module.css";

export default class Login extends React.Component{
    constructor(props){
        super(props);
        this.clickLogin = this.clickLogin.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChangePassWord = this.handleChangePassWord.bind(this);
        this.state={message:"",ID:"",password:""}
    }
    
    clickLogin = ()=>{
        
        if(String(this.state.password).match(/[A-Za-z0-9]+/)==null || String(this.state.ID).match(/[A-Za-z0-9]+/) == null){
            
           return; 
        }
        if(this.state.password == "" || this.state.ID == ""){
            
            return;
        }
        const api = new IkiikiFaceDiagnoseAPI();
        api.callLoginAPI(this.state.ID,this.state.password)
        .then(result =>{
            
            //resultがnull,undifinedな場合にエラーとして処理を行う
            if(!result){
                throw new Error();
            }
            if (result.hasLoginAuthenticated) {
                ReactDOM.render(<FunctionSelection ID={result.ID} />, document.getElementById("root"));                
            }else{
                if(!result.message){
                    throw new Error();
                }
                this.setState({message:result.message});
            }
        }).catch(()=>{
            this.setState({message:"予期しないエラーが発生しました。しばらく待ってから再度実行してください。"});
        });
    }

    handleSubmit = (e)=>{
        e.preventDefault();
    }

    handleChangeID = (e)=>{
        this.setState({ID:e.target.value});
    }

    handleChangePassWord = (e)=>{
        this.setState({password:e.target.value});
    }

    render() {  
        return(
            <>
                <h1>イキイキ顔診断　ログイン画面</h1>  
                <div className={classes.main}>
                    <form onSubmit={this.handleSubmit}>
                        ID<br/>
                        <input type="text" className={classes.box1} placeholder="ID" id="ID" pattern="^[0-9a-zA-Z]+$" title="半角英数字のみ入力してください" required value={this.state.ID} onChange={this.handleChangeID}></input><br/>
                        パスワード<br/>
                        <input type="password" className={classes.box2} placeholder="Password" id="password" pattern="^[0-9a-zA-Z]+$" title="半角英数字のみ入力してください"required value={this.state.password} onChange={this.handleChangePassWord}></input>
                        <button type="submit" onClick={this.clickLogin}>ログイン</button>
                        <div className={classes.errMessage} id="errMsg">
                            <p>{this.state.message}</p>
                        </div>
                    </form>
                </div>  
            </>
        )
    }
}