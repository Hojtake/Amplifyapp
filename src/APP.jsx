import React from "react";
import Login from "./Login";
import FunctionSelection from "./FunctionSelection";
import Diagnosis from "./Diagnosis";

export default class APP extends React.Component{
    constructor(props){
        super(props);
        this.state = {renderComponent : Login.name, ID : "", ikiikiResults:[],hasReadIkiikiResult:false};
        this.selectComponent = this.selectComponent.bind(this);
        this.setID = this.setID.bind(this);
        this.setHasReadIkiikiResult = this.setHasReadIkiikiResult.bind(this);
        this.setIkiikiResults = this.setIkiikiResults.bind(this);
    }
    
    selectComponent = (ComponentName) => {

        this.setState({renderComponent : ComponentName});
    }

    setID(ID){
        this.setState({ID : ID});
    }

    setHasReadIkiikiResult(result){
        this.setState({hasReadIkiikiResult: result});
    }

    setIkiikiResults(ikiikiResults){
        this.setState({ikiikiResults:ikiikiResults})
    }

    render(){
        if(this.state.renderComponent == Login.name){
            return(
                <>   
                    <Login 
                    callback={this.selectComponent} 
                    setID = {this.setID}
                    setIkiikiResults = {this.setIkiikiResults}
                    setHasReadIkiikiResult = {this.setHasReadIkiikiResult} />
                </>
            );
        }else if(this.state.renderComponent == FunctionSelection.name){
            return(
                <>   
                    <FunctionSelection 
                    callback={this.selectComponent} 
                    ID={this.state.ID} 
                    ikiikiResults={this.state.ikiikiResults}
                    hasReadIkiikiResult={this.state.hasReadIkiikiResult}
                    />
                </>
            );
        }else if(this.state.renderComponent == Diagnosis.name){
            return(
                <>   
                    <Diagnosis 
                    callback={this.selectComponent} 
                    ID={this.state.ID}
                    setIkiikiResults = {this.setIkiikiResults}
                    ikiikiResults={this.state.ikiikiResults}/>
                </>
            );
        }else{
            return(
                <>
                エラーが発生しました再度読み込みを行ってください
                </>
            )
        }
    }
}