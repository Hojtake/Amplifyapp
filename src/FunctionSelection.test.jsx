import FunctionSelection from "./FunctionSelection";
import React from "react";
import APP from "./APP";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import userEvent from "@testing-library/user-event";
import Diagnosis from "./Diagnosis";
import IkiikiFaceDiagnoseAPI from "./IkiikiFaceDiagnoseAPI";
import Login from "./Login";

jest.mock("./IkiikiFaceDiagnoseAPI");
const VALID_ID = "testuser";
const VALID_PASSWORD = "testpassword";
it("正常系、過去の診断結果が存在する場合テーブルが作成されていることの確認", async () => {
    const dummyIkiikiResults=[
        {"diagnoseDate":"2022/01/01","ikiikiValue":70 },
        {"diagnoseDate":"2022/01/02","ikiikiValue":50 },
        {"diagnoseDate":"2022/01/03","ikiikiValue":60 }
    ]
    render(
        <FunctionSelection
        ID={VALID_ID}
        ikiikiResults={dummyIkiikiResults}
        hasReadIkiikiResult={true}
        />
    );
    const table = document.querySelector("table");
    expect(table.rows.length).toBe(4);

});

it("正常系、過去の診断結果が存在しない場合、メッセージが表示されることの確認", async () => {

    const dummyIkiikiResults=[];

    render(<FunctionSelection 
        ID={VALID_ID}
        ikiikiResults={dummyIkiikiResults}
        hasReadIkiikiResult={true}
        />
    );
    const message = document.getElementById("nodata_message");
    
    expect(message.innerHTML).toBe("まだイキイキ度が登録されていません。");

});


it("正常系、イキイキ顔診断ボタンを押したときにイキイキ顔診断画面に遷移するかのテスト", async () => {
    const dummyResponseLoginResult = {
        hasLoginAuthenticated: true,
        ID: VALID_ID,
        message: null
    }
    const ikiikiResults=[
        {"diagnoseDate":"2022/01/01","ikiikiValue":70 },
        {"diagnoseDate":"2022/01/02","ikiikiValue":50 },
        {"diagnoseDate":"2022/01/03","ikiikiValue":60 }
    ]
    const dummyResponseReadIkiikiResults = {
        hasReadIkiikiResult:true,
        ikiikiResults:ikiikiResults
    }
    IkiikiFaceDiagnoseAPI.mockImplementation(() => {
        return {
            callLoginAPI: () => {
                return Promise.resolve(dummyResponseLoginResult);
            },
            readIkiikiResultAPI: () =>{
                return Promise.resolve(dummyResponseReadIkiikiResults);
            }
        };
    });
    render(<APP/>);
    const inputID = document.getElementById("ID");
    const inputPassWord = document.getElementById("password");

    userEvent.type(inputID, VALID_ID);
    userEvent.type(inputPassWord, VALID_PASSWORD);
    const loginButton = document.querySelector("button[type='submit']");
    await act(async () => {
        userEvent.click(loginButton);
    });

    const tree = renderer.create(
    <Diagnosis 
    ID={VALID_ID}
    ikiikiResults={ikiikiResults}
    />);
    const diagnoseFaceButton = document.querySelector("button[id='diagnoseFace']");
    await act(async () => {
        userEvent.click(diagnoseFaceButton);
    });
    expect(tree).toMatchSnapshot();
});

it("正常系、イキイキ顔診断ボタンを押したときにイキイキ顔診断画面に遷移するかのテスト", async () => {
    const dummyResponseLoginResult = {
        hasLoginAuthenticated: true,
        ID: VALID_ID,
        message: null
    }
    const ikiikiResults=[
        {"diagnoseDate":"2022/01/01","ikiikiValue":70 },
        {"diagnoseDate":"2022/01/02","ikiikiValue":50 },
        {"diagnoseDate":"2022/01/03","ikiikiValue":60 }
    ]
    const dummyResponseReadIkiikiResults = {
        hasReadIkiikiResult:true,
        ikiikiResults:ikiikiResults
    }
    IkiikiFaceDiagnoseAPI.mockImplementation(() => {
        return {
            callLoginAPI: () => {
                return Promise.resolve(dummyResponseLoginResult);
            },
            readIkiikiResultAPI: () =>{
                return Promise.resolve(dummyResponseReadIkiikiResults);
            }
        };
    });
    render(<APP/>);
    const inputID = document.getElementById("ID");
    const inputPassWord = document.getElementById("password");

    userEvent.type(inputID, VALID_ID);
    userEvent.type(inputPassWord, VALID_PASSWORD);
    const loginButton = document.querySelector("button[type='submit']");
    await act(async () => {
        userEvent.click(loginButton);
    });

    const tree = renderer.create(
    <Diagnosis 
    ID={VALID_ID}
    ikiikiResults={ikiikiResults}
    />);
    const diagnoseFaceButton = document.querySelector("button[id='diagnoseFace']");
    await act(async () => {
        userEvent.click(diagnoseFaceButton);
    });
    expect(tree).toMatchSnapshot();
});

it("正常系、ログアウトボタンを押したときにログイン画面に遷移するかのテスト", async () => {
    const dummyResponseLoginResult = {
        hasLoginAuthenticated: true,
        ID: VALID_ID,
        message: null
    }
    const ikiikiResults=[
        {"diagnoseDate":"2022/01/01","ikiikiValue":70 },
        {"diagnoseDate":"2022/01/02","ikiikiValue":50 },
        {"diagnoseDate":"2022/01/03","ikiikiValue":60 }
    ]
    const dummyResponseReadIkiikiResults = {
        hasReadIkiikiResult:true,
        ikiikiResults:ikiikiResults
    }
    IkiikiFaceDiagnoseAPI.mockImplementation(() => {
        return {
            callLoginAPI: () => {
                return Promise.resolve(dummyResponseLoginResult);
            },
            readIkiikiResultAPI: () =>{
                return Promise.resolve(dummyResponseReadIkiikiResults);
            }
        };
    });
    render(<APP/>);
    const inputID = document.getElementById("ID");
    const inputPassWord = document.getElementById("password");

    userEvent.type(inputID, VALID_ID);
    userEvent.type(inputPassWord, VALID_PASSWORD);
    const loginButton = document.querySelector("button[type='submit']");
    await act(async () => {
        userEvent.click(loginButton);
    });

    const tree = renderer.create(
    <Login
    />);
    const diagnoseFaceButton = document.querySelector("button[id='logout']");
    await act(async () => { 
        userEvent.click(diagnoseFaceButton);
    });
    expect(tree).toMatchSnapshot();
});