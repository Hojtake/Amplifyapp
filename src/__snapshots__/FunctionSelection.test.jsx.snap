// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`正常系、イキイキ顔診断ボタンを押したときにイキイキ顔診断画面に遷移するかのテスト 1`] = `
Array [
  <h1>
    イキイキ顔診断画面
  </h1>,
  <div
    className="username"
  >
    <p
      id="ID"
    >
      ID:
      testuser
    </p>
  </div>,
  <div
    className="return_to_function_select_area"
  >
    <button
      id="clickreturn"
      onClick={[Function]}
      tabIndex={1}
    >
      機能選択画面に戻る
    </button>
  </div>,
  <div
    className="message_area"
  >
    <p
      id="operationMsg"
    >
      ・画像選択ボタンを押してください
    </p>
  </div>,
  <div
    className="display"
  >
    <div
      id="photo_area"
      style={
        Object {
          "marginTop": 0,
        }
      }
    >
      
    </div>
  </div>,
  <div
    className="button_area"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        accept=".png,.jpg,.jpeg"
        className="hidden"
        id="filename"
        onChange={[Function]}
        type="file"
      />
      <button
        className="imageSelect"
        onClick={[Function]}
        tabIndex={2}
        type="submit"
      >
        画像を選択
      </button>
    </form>
    <button
      className="diagnose_button"
      disabled={true}
      id="diagnose_button"
      onClick={[Function]}
      tabIndex={3}
    >
      診断する
    </button>
  </div>,
  <div
    className="diagnose_result_area"
    id="diagnose_result_area"
  >
    <p
      id="resist_day"
    />
    <p
      id="result_msg"
    />
  </div>,
]
`;

exports[`正常系、イキイキ顔診断ボタンを押したときにイキイキ顔診断画面に遷移するかのテスト 2`] = `
Array [
  <h1>
    イキイキ顔診断画面
  </h1>,
  <div
    className="username"
  >
    <p
      id="ID"
    >
      ID:
      testuser
    </p>
  </div>,
  <div
    className="return_to_function_select_area"
  >
    <button
      id="clickreturn"
      onClick={[Function]}
      tabIndex={1}
    >
      機能選択画面に戻る
    </button>
  </div>,
  <div
    className="message_area"
  >
    <p
      id="operationMsg"
    >
      ・画像選択ボタンを押してください
    </p>
  </div>,
  <div
    className="display"
  >
    <div
      id="photo_area"
      style={
        Object {
          "marginTop": 0,
        }
      }
    >
      
    </div>
  </div>,
  <div
    className="button_area"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        accept=".png,.jpg,.jpeg"
        className="hidden"
        id="filename"
        onChange={[Function]}
        type="file"
      />
      <button
        className="imageSelect"
        onClick={[Function]}
        tabIndex={2}
        type="submit"
      >
        画像を選択
      </button>
    </form>
    <button
      className="diagnose_button"
      disabled={true}
      id="diagnose_button"
      onClick={[Function]}
      tabIndex={3}
    >
      診断する
    </button>
  </div>,
  <div
    className="diagnose_result_area"
    id="diagnose_result_area"
  >
    <p
      id="resist_day"
    />
    <p
      id="result_msg"
    />
  </div>,
]
`;

exports[`正常系、ログアウトボタンを押したときにログイン画面に遷移するかのテスト 1`] = `
Array [
  <h1>
    イキイキ顔診断　ログイン画面
  </h1>,
  <div
    className="main"
    style={
      Object {
        "alignSelf": "center",
      }
    }
  >
    <form
      onSubmit={[Function]}
    >
      ID
      <br />
      <input
        className="box1"
        id="ID"
        onChange={[Function]}
        pattern="^[0-9a-zA-Z]+$"
        placeholder="ID"
        required={true}
        title="半角英数字のみ入力してください"
        type="text"
        value=""
      />
      <br />
      パスワード
      <br />
      <input
        className="box2"
        id="password"
        onChange={[Function]}
        pattern="^[0-9a-zA-Z]+$"
        placeholder="Password"
        required={true}
        title="半角英数字のみ入力してください"
        type="password"
        value=""
      />
      <br />
      <button
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        ログイン
      </button>
      <div
        className="errMessage"
        id="errMsg"
      >
        <p>
          
        </p>
      </div>
    </form>
  </div>,
]
`;
